python/lsst/pipe/tasks/finalizeCharacterization.py

```diff
@@ -0,0 +1,693 @@
+        input_ref_dict = butlerQC.get(inputRefs)
+
+        band = butlerQC.quantum.dataId['band']
+        visit = butlerQC.quantum.dataId['visit']
+
+        src_dict_temp = {ref.dataId['detector']: ref
+                         for ref in input_ref_dict['srcs']}
+        calexp_dict_temp = {ref.dataId['detector']: ref
+                            for ref in input_ref_dict['calexps']}
+        isolated_star_cat_dict_temp = {ref.dataId['tract']: ref
+                                       for ref in input_ref_dict['isolated_star_cats']}
+        isolated_star_source_dict_temp = {ref.dataId['tract']: ref
+                                          for ref in input_ref_dict['isolated_star_sources']}
+        # TODO: Sort until DM-31701 is done and we have deterministic
+        # dataset ordering.
+        src_dict = {detector: src_dict_temp[detector] for
+                    detector in sorted(src_dict_temp.keys())}
+        calexp_dict = {detector: calexp_dict_temp[detector] for
+                       detector in sorted(calexp_dict_temp.keys())}
+        isolated_star_cat_dict = {tract: isolated_star_cat_dict_temp[tract] for
+                                  tract in sorted(isolated_star_cat_dict_temp.keys())}
+        isolated_star_source_dict = {tract: isolated_star_source_dict_temp[tract] for
+                                     tract in sorted(isolated_star_source_dict_temp.keys())}
+
+        struct = self.run(visit,
+                          band,
+                          isolated_star_cat_dict,
+                          isolated_star_source_dict,
+                          src_dict,
+                          calexp_dict)
+
+        butlerQC.put(struct.psf_cat,
+                     outputRefs.finalized_psf_cat)
+        butlerQC.put(struct.ap_corr_map_cat,
+                     outputRefs.finalized_ap_corr_map_cat)
+        butlerQC.put(pd.DataFrame(struct.output_table),
+                     outputRefs.finalized_src_table)
+
+    def run(self, visit, band, isolated_star_cat_dict, isolated_star_source_dict, src_dict, calexp_dict):
+        """
+        Run the FinalizeCharacterizationTask.
+
+        Parameters
+        ----------
+        visit : `int`
+            Visit number.  Used in the output catalogs.
+        band : `str`
+            Band name.  Used to select reserved stars.
+        isolated_star_cat_dict : `dict`
+            Per-tract dict of isolated star catalog references.
+        isolated_star_source_dict : `dict`
+            Per-tract dict of isolated star source catalog references.
+        src_dict : `dict`
+            Per-detector dict of src catalog references.
+        calexp_dict : `dict`
+            Per-detector dict of calibrated exposure references.
+
+        Returns
+        -------
+        struct : `lsst.pipe.base.struct`
+            Struct with outputs for persistence.
+        """
+        # We do not need the isolated star table at this time.
```

Comment:
I am envisioning code very similar to this being used in `fgcmcal`, and this is a holdover when I was trying to generalize it and gave up because of subtleties in the multi-band needs to `fgcmcal`.  I'll just clean this up here now because it really is not needed.

Reply:
Oh, but I do use it in tests to confirm that all the indexes are set correctly (which is very important to check since I did mess it up the first time).  I've updated the comment accordingly.